src/pkg/leaderElection.go

```diff
@@ -29,11 +35,9 @@ func RunLeaderElection(client *clientset.Clientset, runnerId opslevel.ID, lockNa
 		},
 	}
 
-	ctx, cancel := context.WithCancel(context.Background())
-	defer cancel()
 	logger := log.With().Str("worker", "leader").Logger()
 
-	leaderelection.RunOrDie(ctx, leaderelection.LeaderElectionConfig{
+	go leaderelection.RunOrDie(ctx, leaderelection.LeaderElectionConfig{
```

Comment:
`RunOrDie` is a blocking function - so we need to background it.  Since the outter function caller no longer does this we do it here.  This allows this function to return errors and properly exit the program if leader election fails.

Reply:
It also means that leader election shuts down properly now